plugins/scaffolder-backend/src/scaffolder/stages/publish/helpers.ts

```diff
@@ -67,3 +68,63 @@ export async function initRepoAndPush({
     remote: 'origin',
   });
 }
+
+type BranchProtectionOptions = {
+  client: Octokit;
+  owner: string;
+  repoName: string;
+  isRetry?: boolean;
+};
+
+export const enableBranchProtectionOnDefaultRepoBranch = async ({
+  repoName,
+  client,
+  owner,
+  isRetry = false,
+}: BranchProtectionOptions): Promise<
+  RestEndpointMethodTypes['repos']['updateBranchProtection']['response']
```

Comment:
Just make this `void` for now imo, we're not using the return value and it cleans things up a. bit.